### DISCUSSION TITLE
segmentatiefout bij import BGTNL

### DISCUSSION THREAD

**Discussion by Robkromwijk** [nlextract/NLExtract#369] [Q&A] [answered]
Hallo,

ik importeer iedere paar maanden de BGTNL via https://api.pdok.nl/lv/bgt/download/v1_0/full/custom m.b.v. NLextract. Dit ging altijd probleemloos tot nu. 
Veel foutmeldingen over foutieve geometrie en aan het einde van het proces de melding:

```shell
/home/rob/git/bronnen/src/bgtnl/../../external/nlextract-1.5.4/bgt/etl/etl.sh: regel 42: 10729 Segmentatiefout         (geheugendump gemaakt) python $STETL_HOME/stetl/main.py -c conf/etl-imgeo-v2.1.1.cfg -a $options_file
``` 
Sommige objectklassen zijn wel geimporteerd, maar de meesten zoals wegdeel niet. Ik heb de volledige log bijgevoegd.
[import_bgtnl.log](https://github.com/nlextract/NLExtract/files/14219969/import_bgtnl.log)

Aan de scripts of instellingen heb ik niets veranderd. Zelfs nlextract-1.5.4. opnieuw gedownload en gebruikt, maar hetzelfde resultaat.

Enig idee of dit aan de data of nlextract. En hoe dit te verhelpen?
Dank

**Comment by justb4** [comment, marked as answer]
Issue gemigreerd naar Discussions. Sinds gisteren vervangt deze de mailing lijst en Gitter/Element Chat. 
In Discussions is plek voor dit soort support vragen. Mocht er echt een probleem in NLExtract zijn, kan alsnog issue worden geopend.

De errors zijn 'gewoon', heeft meestal te maken met 'bogen' geometrie, en zijn vaak ook niet meer geldig, zitten in historie. Zie ook issue hierover:  https://github.com/nlextract/NLExtract/issues/363 en op GeoForum discussie. 'Invalid' is rekbaar begrip.

De log lijkt geheel goed, alleen die Segmentatiefout melding aan het eind...Meestal wijst dit op een geheugenprobleem. De BGT dataset blijft maar groeien.  De postprocessing SQL scripts kunnen dan een dergelijke fout geven. De ETL zelf is 'streaming' (Stetl) dus daar ligt het meestal niet. Daarom was er recente aanpassing in de SQL scripts: https://github.com/nlextract/NLExtract/issues/356. 
Je hebt nog niet de huidige 'master' versie geprobeerd? 1.5.4 is plm jaar oud. Binnenkort 1.5.5 uit te brengen.
